SwrveCommonSDK/src/main/java/com/swrve/sdk/SwrveImp.java

```diff
@@ -933,33 +933,38 @@ protected void unbindAndShutdown() {
         }
     }
 
-    protected void removeCurrentDialog() {
+    protected void removeCurrentDialog(Activity callerActivity) {
```

Comment:
isFinishing is not reliable when checked inside of onDestroy. It is only
reliable inside onPause as far as I know.

On Thu, Mar 19, 2015 at 10:40 AM, dominicmarmionswrve <
notifications@github.com> wrote:

> In SwrveCommonSDK/src/main/java/com/swrve/sdk/SwrveImp.java
> <https://github.com/Swrve/swrve-android-sdk/pull/33#discussion_r26743215>:
>
> > @@ -933,33 +933,38 @@ protected void unbindAndShutdown() {
> >          }
> >      }
> >
> > -    protected void removeCurrentDialog() {
> > +    protected void removeCurrentDialog(Activity callerActivity) {
>
> I think we only need to remove the dialog if the owner activity of the
> dialog is finishing. You can find this out by calling
> dialog.getOwnerActivity().isFinishing(). If its nots finishing then don't
> remove the dialog. This way we don't need to add Activity to the lifecyle
> methods.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Swrve/swrve-android-sdk/pull/33/files#r26743215>.
>